dev-jjjjjeong-bin/Invaders-SDP
# [MAINTENANCE] Restructure Codebase

### **[MAINTENANCE] Restructure Codebase**

#### **Description**
This PR reorganizes the codebase to improve its structure, readability, and maintainability. The changes group related functionalities into cohesive directories, streamline naming conventions, and reduce redundancy.

#### **Changes Made**
- Reorganized folders:
  - Moved core logic to `core/`.
  - Consolidated entities under `entities/` (e.g., player, enemy, collectibles).
  - Centralized HUD components in `hud/`.
  - Grouped inventory-related items into `inventory/`.
  - Renamed `level_design` to `level/` for level-related designs.
  - Consolidated screens into `screen/` for UI components.
  - Created a `sounds/` folder for sound management.
  - Added a `utils/` folder for utility classes and helpers.
  - Introduced a `multiplayer/` folder for two-player mode logic.
- Updated package declarations to match the new structure.
- Updated imports across the project to align with the restructured file paths.

#### **How to Test**
1. Clean and rebuild the project.
2. Run the application and verify:
   - Game launches successfully.
   - All screens, levels, sounds, and functionalities work as expected.

#### **Additional Notes**
- Ensure to delete any cached or compiled files (`out/` directory) before testing.
- Check that new developers can easily navigate the new structure.
- This is a non-functional change, so all functionality should remain intact.
